src/components/ColorPicker.js

```diff
@@ -24,30 +24,26 @@ class ColorPicker extends React.Component {
 
     const {
       classes,
-      editableTrail,
+      trail,
       modifyTrail,
     } = this.props;
 
     const changeColor = (color) => {
       const updateColor = `${color.rgb.r},${color.rgb.g},${color.rgb.b}`
-      modifyTrail(editableTrail.get('id'), {fillColor: updateColor})
+      modifyTrail(trail.get('id'), {fillColor: updateColor})
     }
 
-    const selectedColor = editableTrail.get('fillColor').split(',')
+    const [r,g,b] = trail.get('fillColor').split(',')
 
-    const colorObj = {
-      r: selectedColor[0],
-      g: selectedColor[1],
-      b: selectedColor[2]
-    }
+    const colorObj = {r,g,b}
```

Comment:
Don't need to implement this in this PR, but just as a thought exercise: how could this be restructured to be more re-usable? 

We can imagine a component like ColorPicker that just takes in a color and an `onChange` and shouldn't have to worry about anything else; it can control whether it's expanded or not in it's own state and deal with the hex/rgb conversions, but all the stuff about trails and how to modify them and non-color related concerns would be outside in the TrailForm.

Reply:
:star::star: